server/src/internalClusterTest/java/org/opensearch/search/sort/FieldSortIT.java

```diff
@@ -2389,4 +2389,185 @@ public void testLongSortOptimizationCorrectResults() throws InterruptedException
         }
     }
 
+    public void testSimpleSortsPoints() throws Exception {
```

Comment:
Adding dedicated test case that reproduces the issue all the time

Reply:
I think any case where optimization is not applied will get this `UnSupportedException` ?

Reply:
For new numeric types - yes, the ones we support at the moment (long , double, float, half_float and unsigned_long, ...), the `advance` will be there, but otherwise - yes, `UnSupportedException`